airflow/providers/google/cloud/sensors/pubsub.py

```diff
@@ -170,22 +171,41 @@ def execute(self, context: Context) -> None:
                     subscription=self.subscription,
                     max_messages=self.max_messages,
                     ack_messages=self.ack_messages,
-                    messages_callback=self.messages_callback,
                     poke_interval=self.poke_interval,
                     gcp_conn_id=self.gcp_conn_id,
                     impersonation_chain=self.impersonation_chain,
                 ),
                 method_name="execute_complete",
             )
 
-    def execute_complete(self, context: dict[str, Any], event: dict[str, str | list[str]]) -> str | list[str]:
+    def execute_complete(self, context: Context, event: dict[str, str | list[str]]) -> Any:
         """Return immediately and relies on trigger to throw a success event. Callback for the trigger."""
         if event["status"] == "success":
             self.log.info("Sensor pulls messages: %s", event["message"])
+            if self.messages_callback:
+                received_message_fmt = self._convert_to_received_message(event["message"])
+                message_callback_response = self.messages_callback(received_message_fmt, context)
+
+                return message_callback_response
+
             return event["message"]
         self.log.info("Sensor failed: %s", event["message"])
         raise AirflowException(event["message"])
 
+    def _convert_to_received_message(self, messages: Any):
```

Comment:
```suggestion
    def _convert_received_messages(self, messages: Any) -> list[Any]:
```

Reply:
It looks like returning a list of any, but doesn't look like the same behavior as `self._return_value` 🤔  Could you please elaborate a bit?

Reply:
for the `_convert_received_messages` can updated list[Any],  but for `execute_complete` return type should be Any i feel, If user provides `messages_callback` callable function , they might return different type so have added Any.

Reply:
`self. _return_value` is inside normal poke function , but `_convert_to_received_message` is for trigger event, because the trigger sends back list of dict.

Reply:
got it. I miss the `messages_callback` part 🤦‍♂️